File: frontend/src/types/llmCallLog.ts
// LLM调用记录相关类型定义

// 功能类型枚举
export type LLMCallLogFunctionType =
  | 'ai_grade_homework'
  | 'ai_generate_knowledge_graph'
  | 'ai_create_resource'
  | 'ai_generate_question'
  | 'learning_profile_assessment'
  | 'personalized_learning'
  | 'ai_quiz';

// 状态枚举
export type LLMCallLogStatus = 'success' | 'failed';

// 列表项类型
export interface LLMCallLogListItem {
  id: number;
  function_type: LLMCallLogFunctionType;
  user_id: number;
  user_name: string | null;
  user_role: string;
  llm_config_id: number | null;
  llm_config_name: string | null;
  result_summary: string | null;
  execution_time_ms: number | null;
  status: LLMCallLogStatus;
  created_at: string;
}

// 详情类型
export interface LLMCallLogDetail {
  id: number;
  function_type: LLMCallLogFunctionType;
  user_id: number;
  user_name: string | null;
  user_role: string;
  llm_config_id: number | null;
  llm_config_name: string | null;
  prompt: string;
  result: string | null;
  execution_time_ms: number | null;
  status: LLMCallLogStatus;
  error_message: string | null;
  related_id: number | null;
  related_type: string | null;
  created_at: string;
}

// 列表响应类型
export interface LLMCallLogListResponse {
  items: LLMCallLogListItem[];
  total: number;
  skip: number;
  limit: number;
}

// 功能类型显示名称映射
export const FUNCTION_TYPE_NAMES: Record<LLMCallLogFunctionType, string> = {
  ai_grade_homework: 'AI批改作业',
  ai_generate_knowledge_graph: 'AI生成知识图谱',
  ai_create_resource: 'AI创作教学资源',
  ai_generate_question: 'AI出题',
  learning_profile_assessment: '学习偏好测评',
  personalized_learning: '个性化学习',
  ai_quiz: 'AI测评',
};

File: backend/app/schemas/llm_call_log.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class LLMCallLogListItem(BaseModel):
    """LLM调用记录列表项"""
    id: int
    function_type: str = Field(..., description="调用功能类型")
    user_id: int = Field(..., description="调用用户ID")
    user_name: Optional[str] = Field(None, description="用户姓名")
    user_role: str = Field(..., description="用户角色")
    llm_config_id: Optional[int] = Field(None, description="LLM配置ID")
    llm_config_name: Optional[str] = Field(None, description="LLM配置名称")
    result_summary: Optional[str] = Field(None, description="结果摘要（前100字符）")
    execution_time_ms: Optional[int] = Field(None, description="执行时长（毫秒）")
    status: str = Field(..., description="调用状态：success/failed")
    created_at: datetime = Field(..., description="创建时间")
    
    class Config:
        from_attributes = True


class LLMCallLogDetail(BaseModel):
    """LLM调用记录详情"""
    id: int
    function_type: str = Field(..., description="调用功能类型")
    user_id: int = Field(..., description="调用用户ID")
    user_name: Optional[str] = Field(None, description="用户姓名")
    user_role: str = Field(..., description="用户角色")
    llm_config_id: Optional[int] = Field(None, description="LLM配置ID")
    llm_config_name: Optional[str] = Field(None, description="LLM配置名称")
    prompt: str = Field(..., description="提示词")
    result: Optional[str] = Field(None, description="返回结果")
    execution_time_ms: Optional[int] = Field(None, description="执行时长（毫秒）")
    status: str = Field(..., description="调用状态：success/failed")
    error_message: Optional[str] = Field(None, description="错误信息")
    related_id: Optional[int] = Field(None, description="关联的业务ID")
    related_type: Optional[str] = Field(None, description="关联的业务类型")
    created_at: datetime = Field(..., description="创建时间")
    
    class Config:
        from_attributes = True


class LLMCallLogListResponse(BaseModel):
    """LLM调用记录列表响应"""
    items: list[LLMCallLogListItem]
    total: int = Field(..., description="总记录数")
    skip: int = Field(..., description="跳过记录数")
    limit: int = Field(..., description="每页记录数")
